a3fe/configuration/engine_config.py

```diff
@@ -326,17 +276,96 @@ def get_somd_config(self, run_dir: str, config_name: str = "somd_config") -> str
+            lambda_str = f"[{', '.join(str(x) for x in self.lambda_array)}]"
+            config_lines.append(f"lambda_array = {lambda_str}")
+        if self.lambda_val is not None:
+            config_lines.append(f"lambda_val = {self.lambda_val}")
+
+        # 3) Alchemical files path
+        config_lines.extend(["", "### Alchemical Files ###"])
+        if self.morphfile:
+            config_lines.append(f"morphfile = {self.morphfile}")
+        if self.topfile:
+            config_lines.append(f"topfile = {self.topfile}")
+        if self.crdfile:
+            config_lines.append(f"crdfile = {self.crdfile}")
+
+        # 5) Boresch restraints 
+        if self.boresch_restraints_dictionary is not None:
+            config_lines.extend(["", "### Boresch Restraints Dictionary ###"])
+            config_lines.append(f"boresch restraints dictionary = {self.boresch_restraints_dictionary}")
+
         # Add any extra options
         if self.extra_options:
             config_lines.extend(["", "### Extra Options ###"])
             for key, value in self.extra_options.items():
                 config_lines.append(f"{key} = {value}")
 
         # Write the configuration to a file
-        config_path = _os.path.join(run_dir, f"{config_name}.cfg")
+        config_path = _os.path.join(run_dir, config_filename)
         with open(config_path, "w") as f:
             f.write("\n".join(config_lines) + "\n")
+        
+        return config_path
+
+    @classmethod
+    def _from_config_file(cls, config_path: str) -> "SomdConfig":
+        """Create a SomdConfig instance from an existing configuration file."""
+        with open(config_path, "r") as f:
+            config_content = f.read()
+
+        config_dict = {}
+        for line in config_content.split("\n"):
+            line = line.strip()
+            if line and not line.startswith("#") and "=" in line:
+                key, value = [x.strip() for x in line.split("=", 1)]
+                
+                if key == "lambda array":
+                    value = [float(x.strip()) for x in value.split(",")]
+                elif "*" in value:
+                    value = value.split("*")[0].strip()
+                    try:
+                        value = float(value)
+                    except ValueError:
+                        pass
+                elif value.lower() == "true":
+                    value = True
+                elif value.lower() == "false":
+                    value = False
+                else:
+                    try:
+                        if "." in value:
+                            value = float(value)
+                        elif value.isdigit():
+                            value = int(value)
+                    except ValueError:
+                        pass
+                key = key.replace(" ", "_")
+                config_dict[key] = value
+
+        return cls(**config_dict)
+
+    def write_somd_config(self, output_dir: str, lam: float) -> str:
```

Comment:
Have a think about if we really need this - I think it might be best just to do this in the lambda window/ simulation parts of the code.